lib/scheduler.pm

```diff
@@ -79,22 +80,22 @@ in the same file than the schedule or in a dedicated file only for data.
 sub parse_test_data {
     my ($schedule) = shift;
     $test_data = {};
-    # return if section is not defined
-    return unless exists $schedule->{test_data};
 
-    if (defined(my $import = $schedule->{test_data}->{$include_tag})) {
-        # Allow both lists and scalar value for import "!include" key
-        if (ref $import eq 'ARRAY') {
-            for my $include (@{$import}) {
-                _import_test_data_from_yaml($include);
-            }
-        }
-        else {
-            _import_test_data_from_yaml($import);
-        }
+    # return if test_data section is not defined in schedule file
+    if (exists $schedule->{test_data}) {
+        # import test data using !include from test_data section in schedule file
+        _import_test_data_included($schedule->{test_data});
+
+        # test_data from schedule file has priority over included data
+        $test_data = {%$test_data, %{$schedule->{test_data}}};
+    }
+
+    # import test data directly from data file
+    if (my $yamlfile = get_var('YAML_TEST_DATA')) {
+        # test data from data file has priority over test_data from schedule
+        _import_test_data_from_yaml(path => $yamlfile, allow_included => 1);
     }
-    # test_data from schedule file has priority over imported one
-    $test_data = {%$test_data, %{$schedule->{test_data}}};
+    print Dumper($test_data);
```

Comment:
Let's use `diag` here and not native printing to stdout.

Reply:
I though the same, but reading the docs, it looks like only is for coloring stuff, maybe the inner implementation of diag allows something else...I will change it anyway to use openQA functions but I don't know the advantage.

Reply:
`sub diag` is test api method for logging in openQA. Not sure if printing to stdout will end up in the logs at all. It's not like if there is advantage, but more like it is not guaranteed to work other way =)

Reply:
I had to add `use testapi 'diag';` to each test, let's see if CI passes.

Reply:
That's weird, but I can see that you don't import it here.